### DISCUSSION TITLE
Cannot Update Artists

### DISCUSSION THREAD

**Discussion by seancrossettie** [nss-evening-cohort-14/evening-client-side#277] [Capstones!] [answered]
## Name 
Sean Rossettie

## Issue
I am currently working on updating my artists and cannot get the single key-value pair of my artist to in firebase on the click event. I currently am using a switch case to set the value of the updated artist to an object containing all of the artist's information. Inside of that, I am assigning the kvp of 'favorite' to the opposite of whatever it is based on an if-else statement, but cannot get that to update to firebase.

## Error messages
No error messages appear

## What have you googled?
https://forum.freecodecamp.org/t/how-to-update-state-using-setstate-method-of-the-usestate-hook-in-react-js/388227
https://www.codegrepper.com/code-examples/javascript/how+to+update+a+key+of+a+usestate+object
https://dirask.com/questions/React-how-to-set-state-with-dynamic-key-value-pair-DkBgg1

## What have you tried?
I have tried:
- restarting my server
- using a '!' operator on my 'favorite' kvp and then updating

## Link to File in Repo
https://github.com/seancrossettie/show-finder/blob/favorite-artists/src/components/Artists/MyArtistCards.js

**Comment by TrinityChristiana** [comment]
Remember, setting state is not instantaneous, just like calling a promise. So on a line immediately after setting state, you cannot get the updated value of that state object. You can see this in action by console warning the state object right after setting it. 

I recommend finding another way to create an object with the wanted favorite value to send to your updateArtist function. Update us on how that goes! 😃

**Comment by seancrossettie** [comment, marked as answer]
Figured it out! All I had to do was create a new instance of the artist object by declaring a new object with all the matching kvp's, and then on a click fire a function that used dot notation to update the artist's favorite status. I then used that updated artist object in my update artist function.
